### PR TITLE
feat(TaskPageSprintfilter): Add changeSprint functionality to Task Pa…

### PR DESCRIPTION
**Pull request by Akshay-Jak**
…ge directly

Signed-off-by: Akshay <akshayjani2147@gmail.com>

### Functionality:
A function to change the sprint directly from the Task Page instead of going back to Board page and change

### Solution:
Made a function changeSprint to filter between sprints

### Risk level:
- [ ] high 
- [x] medium
- [ ] low

### How to test:
ng serve
firebase emulators:start

### CODE REVIEW

File: src/app/body/tasks/tasks.component.ts [hunk 1]
```diff
@@ -28,8 +31,12 @@ export class TasksComponent implements OnInit {
   filterDifficulty: string
   filterStatus: string
   filterProject: string
+  filterSprintNumber: number;
   showFilter: boolean = false
-  constructor(private route: ActivatedRoute, private router: Router, private db: AngularFirestore, public navbarHandler: NavbarHandlerService, public authService: AuthService) { }
+
+  teamData: TeamDataId[] = [];
+
```
Comment (arjund-06) [inline src/app/body/tasks/tasks.component.ts:38]: Remove the unnecessary empty lines

File: src/app/body/tasks/tasks.component.ts [hunk 2]
```diff
@@ -91,6 +98,25 @@ export class TasksComponent implements OnInit {
...
+        this.teamData = teams;
+        console.log(1);
+        newSprintNumber = this.teamData[0].CurrentSprintId;
+        this.currentSprintName = this.fullSprintName(newSprintNumber);
+        this.router.navigate(['Tasks/', this.teamId, this.currentSprintName]);
+        this.readData();
+      });
+    } else {
+      this.currentSprintNumber = newSprintNumber;
+      console.log(2);
```
Comment (arjund-06) [inline src/app/body/tasks/tasks.component.ts:113]: Remove the console.logs

**Review by arjund-06 — CHANGES_REQUESTED** [review]
Make the following changes

**Review by arjund-06 — APPROVED** [review]
Ship it

File: src/app/body/tasks/tasks.component.html [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+                <span class="btn dropdown-toggle" type="button" id="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Menu</span>
+                <div class="dropdown-menu dropdown-menu-left" id="dropdownUserCard">
+                    <div class="input-group">
+                        <input class="form-control" style="font-size: 11px;" type="number" id="filterSprintNumberInput" [(ngModel)]="filterSprintNumber">
+                        <div class="col-md-3 col-3">
+                            <button class="btn" id="dropdownButton" style="font-size: 10px;" (click)="changeSprint(filterSprintNumber)">Filter</button>
+                        </div>
+                    </div>
+                </div>
+
```
Comment (viv9k) [inline src/app/body/tasks/tasks.component.html:17]: Let's have this input box inside filter list and also rearrange the list horizontally.

**Review by viv9k — CHANGES_REQUESTED** [review]
Please do the required changes

File: src/app/body/tasks/tasks.component.ts [hunk 4]
```diff
@@ -91,6 +97,23 @@ export class TasksComponent implements OnInit {
...
+      this.applicationSettingsService.getTeamDetails(this.teamId).subscribe(teams => {
+        this.teamData = teams;
+        newSprintNumber = this.teamData[0].CurrentSprintId;
+        this.currentSprintName = this.fullSprintName(newSprintNumber);
+        this.router.navigate(['Tasks/', this.teamId, this.currentSprintName]);
+        this.readData();
+      });
+    } else {
+      this.currentSprintNumber = newSprintNumber;
+      this.currentSprintName = this.fullSprintName(newSprintNumber);
```
Comment (viv9k) [inline src/app/body/tasks/tasks.component.ts:111]: I don't think we are using backlog in link when somebody goes to backlog and same with deleted. So better we keep the number only

**Review by viv9k — CHANGES_REQUESTED** [review]
Please do the required changes

**Review by viv9k — APPROVED** [review]
ship it

**Pull request merged**